File: Models/Contribution.cs
namespace WebAppTrafficSign.Models
{
    public class Contribution
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public int SignId { get; set; } // For Update/Delete actions - reference to existing TrafficSign
        public int TrafficSignId { get; set; }
        public string Action { get; set; } // e.g., "Add", "Update", "Delete"
        public string Description { get; set; }
        public string Status { get; set; } // e.g., "Pending", "Approved", "Rejected"
        public ICollection<Vote> Votes { get; set; } = new List<Vote>();
        public User User { get; set; }
        public TrafficSign TrafficSign { get; set; }
        public string ImageUrl { get; set; }
        public DateTime CreatedAt { get; set; }
        
        // Fields for new traffic signs (Action = "Add")
        public string Type { get; set; } // Type of traffic sign
        public double? Latitude { get; set; } // Location latitude
        public double? Longitude { get; set; } // Location longitude
    }
}

File: Mobile/traffic_sign_mobile/lib/features/contributions/data/models/contribution.dart
class Contribution {
  Contribution({
    required this.id,
    required this.userId,
    required this.action,
    required this.description,
    required this.status,
    required this.createdAt,
    this.type,
    this.latitude,
    this.longitude,
    this.imageUrl,
  });

  factory Contribution.fromJson(Map<String, dynamic> json) {
    return Contribution(
      id: json['id'] as int,
      userId: json['userId'] as int,
      action: json['action'] as String? ?? 'Add',
      description: json['description'] as String? ?? '',
      status: json['status'] as String? ?? 'Pending',
      createdAt: DateTime.parse(json['createdAt'] as String),
      type: json['type'] as String?,
      latitude: (json['latitude'] as num?)?.toDouble(),
      longitude: (json['longitude'] as num?)?.toDouble(),
      imageUrl: json['imageUrl'] as String?,
    );
  }

  final int id;
  final int userId;
  final String action;
  final String description;
  final String status;
  final DateTime createdAt;
  final String? type;
  final double? latitude;
  final double? longitude;
  final String? imageUrl;
}
